core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1055,6 +1080,32 @@ public synchronized void afterMerge(OnGoingMerge merge) {
                     deactivateThrottling();
                 }
             }
+            if (indexWriter.hasPendingMerges() == false && System.nanoTime() - lastWriteNanos >= engineConfig.getFlushMergesAfter().nanos()) {
```

Comment:
so if a merge finishes after the 5 min it will get flushed by this mechanism, if it finishes before then synced flush will flush it I guess?